pkg/docker/runner_int_test.go

```diff
@@ -95,13 +98,92 @@ func TestRun(t *testing.T) {
 	}
 }
 
-func prePullTestImages(t *testing.T) {
+// TestRunDigested ensures that passing a digested image to the runner will deploy
+// that image instead of the previously built one. This test is depended on the
+// specific image since its verifying the function's output.
+func TestRunDigested(t *testing.T) {
+	root, cleanup := Mktemp(t)
+	defer cleanup()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*10)
+	t.Cleanup(cancel)
+
+	image := testImageWithDigest
```

Comment:
Would it be possible to first build the image, get its digested image name, and then use that digested image for testing "func run" (from a new temporary directory such that the build stamp from `.func` is not considered)?  That would slow the test a bit, but would remove the need for a pre-built external container for the test